cashweb-registry/src/http/server.rs

```diff
@@ -218,3 +238,50 @@ async fn handle_get_metadata_range(
             .collect(),
     }))
 }
+
+async fn handle_get_messages(
+    Path(topic): Path<String>,
+    Query(params): Query<MessagesQuery>,
+    Extension(server): Extension<RegistryServer>,
+) -> Result<Protobuf<cashweb_payload::proto::SignedPayloadSet>, HttpRegistryError> {
+    let signed_payloads = server
+        .registry
+        .get_messages(topic.as_str(), params.from, params.to)?
+        .iter()
+        .map(|signed_payload| signed_payload.to_proto())
+        .collect();
+
+    let payload_page = SignedPayloadSet {
+        items: signed_payloads,
+    };
+
+    Ok(Protobuf(payload_page))
+}
+
+async fn handle_get_message(
+    Path(hex_hash): Path<String>,
+    Extension(server): Extension<RegistryServer>,
+) -> Result<Protobuf<cashweb_payload::proto::SignedPayload>, HttpRegistryError> {
+    let payload_hash = hex::decode(&hex_hash).map_err(|err| HttpRegistryError(err.into()))?;
+    let message = server.registry.get_message(payload_hash)?;
+
+    Ok(Protobuf(message.to_proto()))
+}
+
+async fn handle_put_message(
+    Protobuf(message): Protobuf<cashweb_payload::proto::SignedPayload>,
+    Extension(server): Extension<RegistryServer>,
+) -> Result<Protobuf<proto::PutSignedPayloadResponse>, HttpRegistryError> {
+    let result = server
+        .registry
+        .put_message(SignedPayload::parse_proto(&message)?)
+        .await?;
+
+    Ok(Protobuf(proto::PutSignedPayloadResponse {
+        txid: result
+            .txids
+            .into_iter()
+            .map(|txid| txid.as_slice().to_vec())
+            .collect(),
+    }))
```

Comment:
This will need a push to the tokio worker sending to peers. I'll factor this out in another commit though.